ambry-network/src/main/java/com/github/ambry/network/http2/AmbrySendToHttp2Adaptor.java

```diff
@@ -60,6 +72,6 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
       return;
     }
     DefaultHttp2DataFrame dataFrame = new DefaultHttp2DataFrame(byteBufChannel.getBuf(), true);
-    ctx.write(dataFrame, promise);
+    ctx.writeAndFlush(dataFrame, promise);
```

Comment:
I would go against using writeAndFlush here, this is a write method, it should just call ctx.write and let the caller do writeAndFlush.

Reply:
I am a little write about using write() only. 
Did some reading, but still not sure if the caller's writeAndFlush can trigger writeAndFlush through all the pipeline. 
Any drawback to use writeAndFlush() here?

Reply:
as per my understanding the functional difference between write and writeAndFlush is that writeAndFlush writes the data immediately to the socket buffer whereas with write() the flushing to socket buffer does not happen untill a certain number of buffer is accumulated at netty layer which is controlled through setWriteBufferHighWaterMark in ChannelConfig i think. Usually writeAndFlush() is needed for latency sensitive protocols, it may a bit costly if lots of small messages are being sent out using writeAndFlush.

Reply:
Looks at the code, if the caller call writeAndFlush, it will internally 
```
    private void invokeWriteAndFlush(Object msg, ChannelPromise promise) {
        if (invokeHandler()) {
            invokeWrite0(msg, promise);
            invokeFlush0();
        } else {
            writeAndFlush(msg, promise);
        }
    }
```
so, we are safe if only write() used in AmbrySendToHttp2Adaptor.java